fairmq/shmem/Socket.h

```diff
@@ -266,35 +291,31 @@ class Socket final : public fair::mq::Socket
 
     int64_t Receive(std::vector<MessagePtr>& msgVec, int timeout = -1) override
     {
-        int flags = 0;
-        if (timeout == 0) {
-            flags = ZMQ_DONTWAIT;
-        }
-        int elapsed = 0;
+        assert(msgVec.empty());
```

Comment:
I think previously it was fine to receive into a non-empty vector. Is it different now? `reserve` would behave not as expected, but in principle it should work, or?

We had a request from Alice in the past (few years back) to allow receiving into a non-empty vector.

Reply:
I guess, I can just `msgVec.reserve(msgVec.size() + n);` and remove the pre-condition.